mitosheet/mitosheet/parser.py

```diff
@@ -656,48 +717,47 @@ def get_header_index_header_index_match(formula: str, raw_parser_matches: List[R
         formula,
         formula_label,
         string_matches,
-        df
+        dfs,
+        df_names,
+        sheet_index
     )
 
     match_index = 0
     while match_index < len(raw_parser_matches):
-
-        header_index_header_index_match = get_header_index_header_index_match(formula, raw_parser_matches, match_index)
-        if header_index_header_index_match is not None:
+        if (header_index_header_index_match := get_header_index_header_index_match(formula, raw_parser_matches, match_index)) is not None:
             parser_matches.append(header_index_header_index_match)
             match_index += 4
-            continue
-
-        header_header_match = get_header_header_match(formula, raw_parser_matches, match_index)
-        if header_header_match is not None:
+        elif (sheet_header_header_match := get_sheet_header_header_match(formula, raw_parser_matches, match_index)) is not None:
+            parser_matches.append(sheet_header_header_match)
+            match_index += 3
+        elif (header_header_match := get_header_header_match(formula, raw_parser_matches, match_index)) is not None:
             parser_matches.append(header_header_match)
             match_index += 2
-            continue
-
-        header_index_match = get_header_index_match(formula, raw_parser_matches, match_index)
-        if header_index_match is not None:
+        elif (header_index_match := get_header_index_match(formula, raw_parser_matches, match_index)) is not None:
             parser_matches.append(header_index_match)
             match_index += 2
-            continue
-
-        header_match = get_header_match(formula, raw_parser_matches, match_index)
-        if header_match is not None:
+        elif (header_match := get_header_match(formula, raw_parser_matches, match_index)) is not None:
             parser_matches.append(header_match)
             match_index += 1
-            continue
+        else:
+            raise make_invalid_formula_error(
+                formula,
+                'Something went wrong while parsing the formula. Please contact support.',
```

Comment:
I don't think we need to tell them to contact support. Let's just tell them to edit the formula and try again -- sometimes users write wrong formulas, and then fix them!